src/block/Composter.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\utils\CompostFactory;
+use pocketmine\data\runtime\RuntimeDataDescriber;
+use pocketmine\event\block\ComposterEmptyEvent;
+use pocketmine\event\block\ComposterFillEvent;
+use pocketmine\event\block\ComposterMatureEvent;
+use pocketmine\item\Item;
+use pocketmine\item\VanillaItems;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use pocketmine\world\BlockTransaction;
+use pocketmine\world\particle\CropGrowthEmitterParticle;
+use pocketmine\world\sound\ComposterEmptySound;
+use pocketmine\world\sound\ComposterFillSound;
+use pocketmine\world\sound\ComposterFillSuccessSound;
+use pocketmine\world\sound\ComposterReadySound;
+use function max;
+use function mt_rand;
+
+class Composter extends Transparent{
+
+	public const MIN_LEVEL = 0;
+	public const IMMATURE_LEVEL = 7;
```

Comment:
> I think, FULL_LEVEL can be used and for MAX_LEVEL since it only indicates the possibility of the composter dropping a bone meal I think COLLECTING_LEVEL might be better?

so FULL_LEVEL for 7 and COLLECTING_LEVEL for 8?

Reply:
> > I think, FULL_LEVEL can be used and for MAX_LEVEL since it only indicates the possibility of the composter dropping a bone meal I think COLLECTING_LEVEL might be better?
> 
> so FULL_LEVEL for 7 and COLLECTING_LEVEL for 8?

if so, MIN_LEVEL should be renamed to EMPTY_LEVEL?

Reply:
Can be better probably yes.